app/components/autocomplete/at_mention/index.ts

```diff
@@ -19,28 +20,44 @@ import type {WithDatabaseArgs} from '@typings/database/database';
 type OwnProps = {channelId?: string}
 const enhanced = withObservables([], ({database, channelId}: WithDatabaseArgs & OwnProps) => {
     const currentUser = observeCurrentUser(database);
+    const currentTeam = observeCurrentTeam(database);
+    const currentTeamId = observeCurrentTeamId(database);
```

Comment:
Quick review, will check the rest tomorrow.

Two things:
1. With the team you have access to the teamId so no reason why querying this two.

Now the big thing, the team is not really the current team, it should be the team of the channel and fallback to the current team for DM/GM.

Why?
Autocomplete can be opened also from a thread, for example I can search for a post, saved post, recent mention etc.. and open the thread directly from there and they can be from any team.

Reply:
Oo yup very good point - will get it from channel instead and then revert to current team.